### DISCUSSION TITLE
How to enable a custom image picker option in existing UI

### DISCUSSION THREAD

**Discussion by satvikreddy** [fluttercandies/flutter_wechat_assets_picker#222] [Q&A]
First of all, this is an amazing package with beautiful UI. Kudos for achieving this!!!👍

**Problem**: How to provide asset picker with a List of File or List of imageUrl and get picker functionality without overriding UI build methods?

**Use case**: We have a form where user can add image(s) from their phone gallery or their instagram account. The app already has the instagram imageUrl list. We just want to enable picking of images using the same UI that wechats_asset_picker package has built.

What is the simplest/best way to achieve this? Both of the following are okay for our use case.
- launching a standalone picker with instagram pictures (or) a list of images
- showing instagram as an option in the dropdown along with gallery
Can either of these be achieved without having to rewrite UI build methods.

![image](https://user-images.githubusercontent.com/16370824/143002651-a866c9c0-a2ac-4aae-b87f-a2f105957fcb.png)

**Comment by AlexV525** [comment]
The answer is **NO** at present.

The options in the dropdown menu are the concept of `Path`, and the default type of `Path` is `AssetPathEntity`. To insert a custom path into the menu that is outside of the path system, you should at least rewrite the length getter and the widget method that creates each row.
